fluss-client/src/test/java/org/apache/fluss/client/table/FlussTypedClientITCase.java

```diff
@@ -0,0 +1,452 @@
+            Upsert upsert = table.newUpsert();
+            TypedUpsertWriter<AllTypesPojo> writer = upsert.createTypedWriter(AllTypesPojo.class);
+
+            AllTypesPojo p1 = newAllTypesPojo(1);
+            AllTypesPojo p2 = newAllTypesPojo(2);
+            writer.upsert(p1).get();
+            writer.upsert(p2).get();
+
+            // update key 1: change a couple of fields
+            AllTypesPojo p1Updated = newAllTypesPojo(1);
+            p1Updated.str = "a1";
+            p1Updated.dec = new java.math.BigDecimal("42.42");
+            writer.upsert(p1Updated).get();
+            writer.flush();
+
+            // scan as POJOs and verify change types and values
+            TypedLogScanner<AllTypesPojo> scanner =
+                    table.newScan().createTypedLogScanner(AllTypesPojo.class);
+            subscribeFromBeginning(scanner, table);
+
+            List<ChangeType> changes = new ArrayList<>();
+            List<AllTypesPojo> values = new ArrayList<>();
+            while (values.size() < 4) { // INSERT 1, INSERT 2, UPDATE_BEFORE 1, UPDATE_AFTER 1
+                TypedScanRecords<AllTypesPojo> recs = scanner.poll(Duration.ofSeconds(2));
+                for (TypedScanRecord<AllTypesPojo> r : recs) {
+                    changes.add(r.getChangeType());
+                    values.add(r.getValue());
+                }
+            }
+            assertThat(changes)
+                    .contains(ChangeType.INSERT, ChangeType.UPDATE_BEFORE, ChangeType.UPDATE_AFTER);
+            // ensure the last update_after reflects new value
+            int lastIdx = changes.lastIndexOf(ChangeType.UPDATE_AFTER);
+            assertThat(values.get(lastIdx).str).isEqualTo("a1");
+        }
+    }
+
+    @Test
+    void testTypedLookups() throws Exception {
+        Schema schema = allTypesPkSchema();
+        TablePath path = TablePath.of("pojo_db", "lookup_pk");
+        TableDescriptor td = TableDescriptor.builder().schema(schema).distributedBy(2, "a").build();
+        createTable(path, td, true);
+
+        try (Table table = conn.getTable(path)) {
+            TypedUpsertWriter<AllTypesPojo> writer =
+                    table.newUpsert().createTypedWriter(AllTypesPojo.class);
+            writer.upsert(newAllTypesPojo(1)).get();
+            writer.upsert(newAllTypesPojo(2)).get();
+            writer.close();
+
+            // primary key lookup using Lookuper API with POJO key
+            TypedLookuper<PLookupKey> lookuper =
+                    table.newLookup().createTypedLookuper(PLookupKey.class);
+            RowType tableSchema = table.getTableInfo().getRowType();
+            RowToPojoConverter<AllTypesPojo> rowConv =
+                    RowToPojoConverter.of(AllTypesPojo.class, tableSchema, tableSchema);
+
+            LookupResult lr = lookuper.lookup(new PLookupKey(1)).get();
+            AllTypesPojo one = rowConv.fromRow(lr.getSingletonRow());
+            assertThat(one.str).isEqualTo("s1");
```

Comment:
```
assertThat(one).isEqualTo(newAllTypesPojo(1));
```

After adding `equals` and `hashcode` method to the `AllTypesPojo` class, we can simply assert the full record, this can check the full POJO record deserialization.